src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -140,13 +126,43 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
     relations.head
   }
 
+  private def hasParquetAsSourceFormatProperty(
+      relation: FileBasedRelation): Option[(String, String)] = {
+    if (relation.hasParquetAsSourceFormat) {
+      Some(IndexConstants.HAS_PARQUET_AS_SOURCE_FORMAT_PROPERTY -> "true")
+    } else {
+      None
+    }
+  }
+
+  private def hasLineageProperty(spark: SparkSession): Option[(String, String)] = {
+    if (hasLineage(spark)) {
+      Some(IndexConstants.LINEAGE_PROPERTY -> "true")
+    } else {
+      None
+    }
+  }
+
+  private def usesNestedFieldsProperty(indexConfig: IndexConfig): Option[(String, String)] = {
+    if (SchemaUtils.hasNestedFields(indexConfig.indexedColumns ++ indexConfig.includedColumns)) {
+      Some(IndexConstants.USES_NESTED_FIELDS_PROPERTY -> "true")
+    } else {
+      None
+    }
+  }
+
   private def resolveConfig(
       df: DataFrame,
       indexConfig: IndexConfig): (Seq[String], Seq[String]) = {
     val spark = df.sparkSession
-    val dfColumnNames = df.schema.fieldNames
-    val indexedColumns = indexConfig.indexedColumns
-    val includedColumns = indexConfig.includedColumns
+    // Flatten will transform nested field names from `struct(nested, struct(nst, struct(field1)))`
+    // into `Seq("nested.nst.field1")`.
+    val dfColumnNames = SchemaUtils.flatten(df.schema)
+    // The index config will contain the field names as they are given with `.` (dots).
+    // (ie: `Seq("nested.nst.field")`) and they need to be escaped to `nested__nst__field1` as
+    // the index entry know to work with escaped values only.
+    val indexedColumns = SchemaUtils.unescapeFieldNames(indexConfig.indexedColumns)
```

Comment:
What if a column is not nested and has double underscore in its name?

Reply:
The `flatten` method encloses with backticks any field names containing `__` and `unescapeFieldName` checks if the field name is backticks enclosed.